### DISCUSSION TITLE
Cloudflare Pages: POST method handler always triggers error about multiple stream branches

### DISCUSSION THREAD

**Discussion by supernes** [honojs/hono#2294] [Q&A]
Hey everyone, I'm having some trouble with a very basic script and wanted to see if I'm doing something wrong before filing a bug report.

I have a CF Pages project set up with Vite for the frontend, and I'm using Hono for routing in the CF Function (Worker). I'm testing it locally with `wrangler pages dev`:

```typescript
// functions/api/[[route]].ts

import { Hono } from 'hono'
import { handle } from 'hono/cloudflare-pages'

type Bindings = {
    DB: D1Database
}

const api = new Hono<{ Bindings: Bindings }>().basePath('/api');

api.get("/foods", async (ctx) => {
    const resp = await ctx.env.DB.prepare('...').all();

    return ctx.json(resp);
});

api.post("/foods", async (ctx) => {
    // const pform = await ctx.req.parseBody();
    // console.log(pform);

    return ctx.json({ status: "ok" });
})

export const onRequest = handle(api);
```

GET requests work without issue. The problem is that every time the POST handler executes, I get an error in the `wrangler` console:

> [ERROR] Your worker created multiple branches of a single stream (for instance, by calling `response.clone()` or `request.clone()`) but did not read the body of both branches. This is wasteful, as it forces the system to buffer the entire stream of data in memory, rather than streaming it through. This may cause your worker to be unexpectedly terminated for going over the memory limit. If you only meant to copy the request or response headers and metadata (e.g. in order to be able to modify them), use the appropriate constructors instead (for instance, `new Response(response.body, response)`, `new Request(request)`, etc).

The error appears **twice** if I uncomment the lines parsing and logging the request body. 
Tested with POST requests both from the browser and cURL.

Is there a problem with the way I've set things up or is this unintentional behavior?

|   | version |
-------|------
| hono | 4.0.8 |
| wrangler | 3.30.0 |
| Node | v20.10 |

**Comment by yusukebe** [comment]
Hi @supernes 

I can't reproduce it. If you want to have help, please share the minimal project, such as on GitHub, to reproduce it.

**Reply by supernes** [reply]
Thanks for the quick response. Here's a very simple project that demonstrates the issue:

https://github.com/supernes/vite-hono-post

**Reply by yusukebe** [reply]
@supernes 

Thanks! The command seems to be wrong:

```json
"dev:wrangler": "wrangler pages dev --local --port 4567 dist/"
```

That should be:

```json
"dev:wrangler": "wrangler pages dev --local --port 4567 ."
```

**Reply by supernes** [reply]
I believe the command is correct (as long as you run `npm run build` first). 

If you don't want to build the frontend of the demo project statically, you can also try running it with 

```
npx wrangler pages dev --local --port 4567 -- vite
``` 

I've tested it with both invocations and that's not where the problem is.

The worker compiles and runs successfully. The server correctly routes requests to the file containing Hono code, however reading the `req` object or sending a response through the context each trigger warnings from wrangler.

**Reply by yusukebe** [reply]
Hmm. Anyway, it's not Hono matter. The error occurs with a vanilla script:

```ts
export function onRequest() {
  return Response.json({ status: 'ok' })
}
```

**Reply by supernes** [reply]
Thanks for looking into this. I'll do more tests and report if I find anything relevant.
